tests/background_network_workload_test.py

```diff
@@ -0,0 +1,245 @@
+UID = 'name0'
+
+CONFIG_WITH_BACKGROUND_NETWORK = """
+ping:
+    description: Benchmarks ping latency over internal IP addresses
+    vm_groups:
+      vm_1:
+        vm_spec:
+          GCP:
+            machine_type: n1-standard-1
+      vm_2:
+        vm_spec:
+          GCP:
+            background_network_mbits: 300
+            machine_type: n1-standard-1
+"""
+
+CONFIG_WITH_BACKGROUND_NETWORK_BAD_IPFLAG = """
+ping:
+    description: Benchmarks ping latency over internal IP addresses
+    vm_groups:
+      vm_1:
+        vm_spec:
+          GCP:
+            machine_type: n1-standard-1
+            background_network_mbits: 200
+            background_network_iptype: BOTH
+      vm_2:
+        vm_spec:
+          GCP:
+            background_network_mbits: 300
+            machine_type: n1-standard-1
+"""
+
+CONFIG_WITH_BACKGROUND_NETWORK_IPFLAG = """
+ping:
+    description: Benchmarks ping latency over internal IP addresses
+    vm_groups:
+      vm_1:
+        vm_spec:
+          GCP:
+            machine_type: n1-standard-1
+      vm_2:
+        vm_spec:
+          GCP:
+            background_network_mbits: 300
+            background_network_iptype: INTERNAL
+            machine_type: n1-standard-1
+"""
+
+
+
+class TestBackgroundNetworkWorkload(unittest.TestCase):
+
+  def setupCommonFlags(self, mock_flags):
+    mock_flags.os_type = benchmark_spec.DEBIAN
+    mock_flags.cloud = providers.GCP
+
+  def _CheckVMFromSpec(self, spec, num_working):
+    vm0 = spec.vms[0]
+    with patch.object(vm0.__class__, 'RemoteCommand'):
+      with patch.object(vm0.__class__, 'Install'):
+        with patch.object(vm0.__class__, 'AllowPort'):
+          expected_install_post_prepare = num_working
+          # one call for client, one call for server, one call for each pid
+          expected_remote_post_start = 4 * num_working
+          # all the start calls, plus stop server and stop client
+          expected_remote_post_stop = (expected_remote_post_start +
+                                       2 * num_working)
+          side_effect_list = []
+          for i in range(expected_remote_post_stop):
+            side_effect_list.append((str(i), ' '))
+          vm0.RemoteCommand.side_effect = side_effect_list
+          spec.Prepare()
+          self.assertEqual(vm0.Install.call_count,
+                           expected_install_post_prepare)
+          spec.StartBackgroundWorkload()
+          self.assertEqual(vm0.RemoteCommand.call_count,
+                           expected_remote_post_start)
+          self.assertEqual(vm0.AllowPort.call_count,
+                           num_working)
+          spec.StopBackgroundWorkload()
+          self.assertEqual(vm0.RemoteCommand.call_count,
+                           expected_remote_post_stop)
+
+  def testWindowsVMCausesError(self):
+    """ windows vm with background_network_mbits raises exception """
+    with mock_flags.PatchFlags() as mocked_flags:
+      self.setupCommonFlags(mocked_flags)
```

Comment:
Since you're using the two lines above in every test function, consider using mock_flags.PatchTestCaseFlags in the class setUp function instead.